iceoryx_hoofs/concurrent/buffer/include/iox/detail/spsc_sofi.inl

```diff
@@ -65,74 +92,59 @@ inline bool SpscSofi<ValueType, CapacityValue>::setCapacity(const uint64_t newSi
+    // ^     ^
+    // w=3  r=1
+    // 1. The consumer thread loads m_writePosition => 3
+    // |----|--B--|--C--|
+    // ^     ^
+    // w=3  r=1
+    // 2. The producer thread pushes two times
+    // |--D--|--E--|-----|
+    // ^           ^
+    // r=3        w=5
+    // 3. The consumer thread loads m_readPosition => 3. The pop method returns false
+    // => Whereas the queue was full, pop returned false giving the impression that the queue if empty
     uint64_t currentReadPosition = m_readPosition.load(std::memory_order_acquire);
     uint64_t nextReadPosition{0};
 
     bool popWasSuccessful{true};
+
     do
     {
+        // SYNC POINT READ: m_data
+        // See explanation of the corresponding synchronization point in push()
         if (currentReadPosition == m_writePosition.load(std::memory_order_acquire))
         {
             nextReadPosition = currentReadPosition;
             popWasSuccessful = false;
         }
         else
         {
-            // we use memcpy here, since the copy assignment is not thread safe in general (we might have an overflow in
-            // the push thread and invalidates the object while the copy is running and therefore works on an
-            // invalid object); memcpy is also not thread safe, but we discard the object anyway and read it
-            // again if its overwritten in between; this is only relevant for types larger than pointer size
-            // assign the user data
+            // we use memcpy here, to ensure that there is no logic in copying the data
             std::memcpy(&valueOut, &m_data[currentReadPosition % m_size], sizeof(ValueType));
-
-            /// @brief first we need to peak valueOut if it is fitting the condition and then we have to verify
-            ///        if valueOut is not am invalid object, this could be the case if the read position has
-            ///        changed
-            if (m_readPosition.load(std::memory_order_relaxed) == currentReadPosition && !verificator(valueOut))
-            {
-                popWasSuccessful = false;
-                nextReadPosition = currentReadPosition;
-            }
-            else
-            {
-                nextReadPosition = currentReadPosition + 1U;
-                popWasSuccessful = true;
-            }
+            nextReadPosition = currentReadPosition + 1U;
+            popWasSuccessful = true;
         }
 
-        // compare and swap
-        // if(m_readPosition == currentReadPosition)
-        //     m_readPosition = l_next_aba_read_pos
-        // else
-        //     currentReadPosition = m_readPosition
-        // Assign m_aba_read_p to next readable location
+        // We need to check if m_readPosition hasn't changed otherwise valueOut might be corrupted
+        // While memory synchronization is not needed for m_readPosition we need to have a
+        // corresponding m_readPosition.store(release) to the m_readPosition.load(acquire) in the
+        //  push method
```

Comment:
No, `acquire` is required in order to not reorder the memcpy after the CAS operation. With `relaxed` memory order it would be perfectly fine for the CPU/compiler to perform the CAS Operation and do the memcpy only after the CAS operation was successful in order to not do useless memcpy's

Reply:
You're right. Changed